rules-tests/CodeQuality/Rector/Property/TypedPropertyFromToOneRelationTypeRector/Fixture/doctrine_many_to_one.php.inc

```diff
@@ -39,7 +39,7 @@ class DoctrineManyToOne
      * @ORM\ManyToOne(targetEntity="Rector\Doctrine\Tests\CodeQuality\Rector\Property\TypedPropertyFromToOneRelationTypeRector\Source\Company", inversedBy="userBuildings")
      * @ORM\JoinColumn(name="company_id", referencedColumnName="id", nullable=false)
      */
-    private ?\Rector\Doctrine\Tests\CodeQuality\Rector\Property\TypedPropertyFromToOneRelationTypeRector\Source\Company $company3 = null;
+    private \Rector\Doctrine\Tests\CodeQuality\Rector\Property\TypedPropertyFromToOneRelationTypeRector\Source\Company $company3;
```

Comment:
I think this would crash, as this property is not defined via constructor.

https://3v4l.org/lO6Kg

Reply:
yes, that's true. But isn't this than the issue of the framework or the usage? 

So yeah, if the field is `nullable: false` i would say the property itself should also be not null. Doctrine will load the property correctly. if someone will use a partial load of a entity this will also break. 

But what is the alternative? Mark the properties as nullable and look for getter and setter and mark them as not nullable?

or what do you think about an configurable option for that?

Reply:
The code has to be valid after migration, and we have to take context of use into an account. 

Here e.g. the property would have to be set at least in the constructor to make it typed.

In case constructor is present and property is set, the `?` can be removed. I think best way would be to make this optionally configurable as you suggest 👍

Reply:
first time i used this configured rules :D 
Hopefully i implemented them correct. Sadly i didn't find a good way to show it in the automated docs which option is available, which one is the default behavior and how the configuration will change the output.
I only added two examples for the Rule and add a new fixture folder for the configured tests